glui.cpp

```diff
@@ -210,7 +264,17 @@ void GLUI_Main::setup_default_glut_callbacks( void )
   glutPassiveMotionFunc( glui_passive_motion_func );
   glutEntryFunc( glui_entry_func );
   glutVisibilityFunc( glui_visibility_func );
-  /*  glutIdleFunc( glui_idle_func );    // FIXME!  100% CPU usage!      */
+  /*  glutIdleFunc( glui_idle_func );    // FIXME!  100% CPU usage!      */  
+  //This is needed to host the X11 copy-paste selection-owner.
+  //It's really no big deal. The scheduler will divide time up
+  //and unless "vsync" is off a continuously rendering client
+  //will wait for the "vertical-blank."
+  //GLUT isn't a good fit for passive applications, and it is
+  //historically for full-motion 3-D displays.
+  //Clients can figure out a wait solution in their Idle function.
+  //freeglut provides a default idle, which it does not expose,
+  //but it's open-souce and so copy-friendly.
+  glutIdleFunc( glui_idle_func );
```

Comment:
GLUI can't make these kinds of decisions for users. It's the user's job to accommodate the end-user. The librarian's job is to accommodate the user.

The only alternatives are A) a dumb opt-in API for enabling copy-paste. A PITA to maintain, and document, and clumsy. B) Run the SelectionRequest event queue in a secondary thread at all times. Seems like overkill for such a feature. And would require resource management (locking) and so a PITA to maintain. Even if there were already reasons to have a thread running in the background, it wouldn't fundamentally solve the problem that what a program does when it is idling is the user's business. Idle code in libraries is virtually unheard of, because libraries are never in the position to make that decision. At best a kind of portable "yield" function can be provided for users to use if they want to.

I'm not sure running GLUT applications on a battery is ever a good idea. They just aren't that class of application. It's like leaving a video game running. It shouldn't be surprising if it drains your battery. And if it does you don't blame GLUI, you blame the client-user. (The application.)